Source/ACE.Server/Command/Handlers/SentinelCommands.cs

```diff
@@ -220,27 +221,60 @@ public static void BuffPlayers(IEnumerable<Player> players, Player caster, bool
+                uint spellID = (uint)Enum.Parse(tySpell, spellNamPrefix + ((isBane) ? string.Empty : SelfOrOther) + maxSpellLevel);
                 var buffMsg = BuildBuffMessage(spellID);
-                if (buffMsg != null) buffMessages.Add(buffMsg);
+                if (buffMsg != null)
+                {
+                    buffMsg.Bane = isBane;
+                    buffMessages.Add(buffMsg);
+                }
             }
             // buff each player
             players.ToList().ForEach(targetPlayer =>
             {
-                // bake player into the messages
-                buffMessages.ForEach(k => k.SetTargetPlayer(targetPlayer));
-                // update client-side enchantments
-                targetPlayer.Session.Network.EnqueueSend(buffMessages.Select(k => k.SessionMessage).ToArray());
-                // run client-side effect scripts, omitting duplicates
-                targetPlayer.CurrentLandblock?.EnqueueBroadcast(targetPlayer.Location, buffMessages.GroupBy(m => m.SpellBase.TargetEffect).Select(a => a.First().LandblockMessage).ToArray());
-                // update server-side enchantments
-                targetPlayer.EnchantmentManager.AddRange(buffMessages.Select(k => k.Enchantment), caster);
+                if (buffMessages.Any(k => !k.Bane))
+                {
+                    // bake player into the messages
+                    buffMessages.Where(k => !k.Bane).ToList().ForEach(k => k.SetTargetPlayer(targetPlayer));
+                    // update client-side enchantments
+                    targetPlayer.Session.Network.EnqueueSend(buffMessages.Where(k => !k.Bane).ToList().Select(k => k.SessionMessage).ToArray());
+                    // run client-side effect scripts, omitting duplicates
+                    targetPlayer.CurrentLandblock?.EnqueueBroadcast(targetPlayer.Location, buffMessages.Where(k => !k.Bane).ToList().GroupBy(m => m.SpellBase.TargetEffect).Select(a => a.First().LandblockMessage).ToArray());
+                    // update server-side enchantments
+                    targetPlayer.EnchantmentManager.AddRange(buffMessages.Where(k => !k.Bane).ToList().Select(k => k.Enchantment), caster);
+                }
+                if (buffMessages.Any(k => k.Bane))
+                {
+                    // Impen/bane targeted at a player
+                    var items = (targetPlayer as Player).GetAllWieldedItems();
+                    var itembuffs = buffMessages.Where(k => k.Bane).ToList();
+                    foreach (var itemBuff in itembuffs)
+                    {
+                        foreach (var item in items)
+                        {
+                            if (item.WeenieType == WeenieType.Clothing || item.IsShield)
+                            {
+                                itemBuff.SetLandblockMessage(item.Guid);
+                                var enchantmentStatus = targetPlayer.ItemMagic(item, itemBuff.SpellBase, itemBuff.Spell, caster);
```

Comment:
I think it would be better to call the code, CreatePlayerSpell() that is already setup to handle the inner magic routines, rather than changing ItemMagic() from protected to public.  Mostly to make sure something isn't missed.  No need to reinvent the wheel.

Reply:
Maybe.  Does it allow for the suppression of chat messages?  Can I add another parameter for such a feature?

Reply:
I tried this and it causes the target player(s) to go into combat mode and do a casting animation, and this is unwanted in this use case.  The intention for this command is for it to function as transparently as possible.  It will be doing up to 98 of these invocations per player, which is the intention.  If Player.CreatePlayerSpell() could have some options to disable some of the normal behavior it could be used for this.

Reply:
My vote would be to create a new public API call in Player_Magic to handle interfacing to the magic classes, rather than exposing them unnecessarily, somewhat similar to what I did for Item cast spells, with CastItemSpell().  This would also keep core magic functionality contained to a small footprint, rather than having it spread out into other locations.